test/index.js

```diff
@@ -1902,4 +1902,37 @@ describe('normalizr', function () {
     }).should.throw();
   });
 
-});
+  it('can normalize iterables keyed with their id', function () {
+    var user = new Schema('users', {
+      idAttribute: function(obj, key) {
+        return key;
+      }
+    })
+
+    var input = {
+      1: {
+        name: 'Adam'
+      },
+      4: {
+        name: 'Jeremy'
+      }
+    };
+
+    Object.freeze(input);
+
+    normalize(input, arrayOf(user)).should.eql({
```

Comment:
Well, right now the functionality that the previous author wrote is baked into `visitIterable()` (line 48). So like `valuesOf()`, this function would create a `new IterableSchema` anyways. I do agree that `valuesOf()` makes more sense, and perhaps `keyedValuesOf()` more. What do you think about a `keyedValuesOf()` that's yet another alias of `arrayOf()`? Not sure if that's unnecessary API complication, though.